changelog.d/14750.feature

```diff
@@ -0,0 +1 @@
+Support [RFC7636](https://datatracker.ietf.org/doc/html/rfc7636) Proof Key for Code Exchange for OAuth single sign-on.
```

Comment:
Out of curiosity, this PR implements the logic for the authorization endpoint side of PKCE. Where is the token endpoint side of it, i.e. the part where we send `code_verifier` back to the OIDC provider when converting an auth code to a token?

Reply:
@dkasak The changes to `_exchange_code` implement sending the `code_verifier` back:

https://github.com/matrix-org/synapse/blob/5e910b8d76e539fb9e2c02799cd9c258395f9c96/synapse/handlers/oidc.py#L714-L715

From my reading of the RFC this is all that's necessary! Let me know if there's something else to do!

Reply:
Yeah, that should be all, I just managed to completely miss those two lines. Thanks for pointing it out.

Reply:
No problem! 👍